src/routes/synapse/room_events.rs

```diff
@@ -0,0 +1,612 @@
+use std::collections::HashMap;
+
+use actix_web::{
+    put,
+    web::{self, Data},
+    HttpMessage, HttpRequest, HttpResponse,
+};
+use serde::{Deserialize, Serialize};
+use sqlx::{Postgres, Transaction};
+use uuid::Uuid;
+
+use crate::{
+    components::{
+        app::AppComponents,
+        database::{DatabaseComponent, DatabaseComponentImplementation},
+        synapse::{RoomMembersResponse, SynapseComponent},
+    },
+    entities::{
+        friendship_history::{FriendshipHistory, FriendshipHistoryRepository},
+        friendships::{Friendship, FriendshipRepositoryImplementation, FriendshipsRepository},
+    },
+    middlewares::check_auth::{Token, UserId},
+    routes::v1::error::CommonError,
+};
+
+use super::errors::SynapseError;
+
+#[derive(Deserialize, Serialize)]
+pub struct RoomEventResponse {
+    pub event_id: String,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct RoomEventRequestBody {
+    pub r#type: FriendshipEvent,
+}
+
+#[derive(Deserialize, Serialize, PartialEq, Eq, Debug, Clone, Copy, Hash)]
+pub enum FriendshipEvent {
+    #[serde(rename = "request")]
+    REQUEST, // Send a friendship request
+    #[serde(rename = "cancel")]
+    CANCEL, // Cancel a friendship request
+    #[serde(rename = "accept")]
+    ACCEPT, // Accept a friendship request
+    #[serde(rename = "reject")]
+    REJECT, // Reject a friendship request
+    #[serde(rename = "delete")]
+    DELETE, // Delete an existing friendship
+}
+
+lazy_static::lazy_static! {
```

Comment:
Instead of using `lazy_static` and this hash map in this way. What do you think about refactoring this to something like below or similar to this?: 
```rust
pub enum FriendshipState {
    ACCEPT,
    REQUEST,
    CANCEL,
    DELETE,
}

impl FriendshipState {
    pub fn next(&self) -> Self {
        match *self {
            FriendshipState::ACCEPT => FriendshipState::CANCEL,
            FriendshipState::REQUEST => FriendshipState::DELETE,
            FriendshipState::CANCEL => FriendshipState::DELETE,
            FriendshipState::DELETE => FriendshipState::DELETE,
        }
    }
}

// ..OR

impl FriendshipState {
    pub fn next(&self) -> Vec<FriendshipState> {
        match *self {
            FriendshipState::ACCEPT => FriendshipState::CANCEL,
            FriendshipState::REQUEST => FriendshipState::DELETE,
            FriendshipState::CANCEL => FriendshipState::DELETE,
            FriendshipState::DELETE => FriendshipState::DELETE,
        }
    }
}


```

Reply:
That is a valid solution, but the implementation will become too long, for something as simple as a hashmap, why would we want that?